napari/layers/points/points.py

```diff
@@ -450,6 +456,7 @@ def data(self, data: Optional[np.ndarray]):
                             self._face._remove(
                                 np.arange(len(data), len(self._face.colors))
                             )
+                        self._shown = self._shown[: len(data)]
```

Comment:
here is a place when setting empty data will cause the same error as described

Reply:
I don't think this should ever happen after the inconsistent instantiation state. I tried setting to empty data and it doesn't seem to cause issues. At most it complains about operand broadcasting when it can't do it, which is what we want!